src/http_handlers.cpp

```diff
@@ -899,12 +899,13 @@ void ImpuRegDataTask::process_received_reg_data()
 
   if (_type == RequestType::REG)
   {
-    // This message was based on a REGISTER request from Sprout. Check the
-    // cached subscriber state to determine whether this is an initial
-    // registration or a re-registration. If this subscriber is already
-    // registered but is registering with a new binding, we still need to tell
-    // the HSS.
-    if ((reg_state == RegistrationState::REGISTERED) && (!new_binding))
+    // This message was based on a REGISTER request from Sprout. We need to
+    // send an SAR to the HSS unless ALL of the following are true...
+    // - this is a re-registration
+    // - this is not a new binding
+    // - the request hasn't forbidden us from using cached data
+    // - the existing record hasn't timed out.
```

Comment:
I don't understand why you've changed this comment. It doesn't reflect this if check - it reflects a check further down.

Reply:
Because the previous comment was unclear and untrue?   The comment now refers to the whole of the following block of code rather than just the first test, but at least it's accurate?

Reply:
I think the previous statement was true, but I think your comment is now confusing - with it right next to the if block, it looks like it's describing the if block test, but that's not true.

Reply:
I don't really care enough to argue about it :-(   The previous comment wasn't terrible.  I felt I'd improved it slightly! 
 The previous comment said that if this subscriber is already registered but is registering with a new binding then we still need to tell the HSS.   But that's not true.   We _might_ need to tell the HSS in that case.   Whether we _actually_ need to tell the HSS is also dependent on whether we're allowed to use cached data and whether the existing record has timed out.   It seems unnecessarily restrictive to limit comments to only being allowed to describe what's implemented in a single if test -- surely it's often obvious what a single if test is doing, and more helpful to describe what the higher level logic / goal of a chunk of code is?   I agree the comment would be misleading if it said "first check that X, Y, Z and G" if the next line only checked "X and Y".  Oh look -- I seem to  have argued about it anyway :-)